Classes/Dimension/ContentDimension.php

```diff
@@ -14,160 +10,121 @@
  * source code.
  */
 
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\DimensionSpace\Dimension;
+
+use Neos\ContentRepository\DimensionSpace\Dimension\Exception\ContentDimensionValuesAreMissing;
 use Neos\Utility\Arrays;
 
 /**
  * The content dimension domain model
  */
 final class ContentDimension
 {
-    /**
-     * @var ContentDimensionIdentifier
-     */
-    protected $identifier;
+    public readonly ContentDimensionIdentifier $identifier;
 
     /**
      * An array of dimension values, indexed by their identifiers
      *
-     * @var array|ContentDimensionValue[]
+     * @var array<string,ContentDimensionValue>
      */
-    protected $values = [];
+    public readonly array $values;
```

Comment:
Another Value Object would make sense here, to avoid nasty bugs if the array is not properly indexed

Reply:
Agreed, makes a lot of sense